### PR TITLE
Remove all traces of Hamcrest

### PR DESCRIPTION
**Pull request by dkhalanskyjb**
Did it in a semi-automated manner, doing minimal fixes over the output of the command:
```sh
git grep hamcrest |
cut -d: -f1 |
xargs sed -iE -e 's/assertThat(\(.*\),\s*IsEqual(\(.*\)))/assertEquals(\2, \1)/' \
-e 's/assertThat(\(.*\),\s*IsInstanceOf(\(.*\)::class.java))/assert(\1 is \2)/' \
-e 's/assertThat(\(.*\),\s*IsNull())/assertEquals(null, \1)/' \
-e 's/assertThat(\(.*\),\s*IsNot(IsEqual(\(.*\))))/assert(\1 != \2)/' \
-e '/import.*hamcrest/d'
```

There are still some problems with style: in some cases `org.junit.Assert.*` are used, while in some others the native versions provided by Kotlin are.

### CODE REVIEW

File: reactive/kotlinx-coroutines-reactor/test/FluxTest.kt [hunk 1]
```diff
@@ -23,7 +22,7 @@ class FluxTest : TestBase() {
         expect(2)
         flux.subscribe { value ->
             expect(5)
-            Assert.assertThat(value, IsEqual("OK"))
+            Assert.assertEquals("OK", value)
```
Comment (elizarov) [inline reactive/kotlinx-coroutines-reactor/test/FluxTest.kt]: `Assert.` prefix is wrong and should be removed.

File: kotlinx-coroutines-core/jvm/test/channels/ConflatedBroadcastChannelNotifyStressTest.kt [hunk 2]
```diff
@@ -5,10 +5,9 @@
 package kotlinx.coroutines.channels
 
 import kotlinx.coroutines.*
-import org.hamcrest.MatcherAssert.*
-import org.hamcrest.core.*
 import org.junit.*
 import java.util.concurrent.atomic.*
+import kotlin.test.assertEquals
```
Comment (elizarov) [inline kotlinx-coroutines-core/jvm/test/channels/ConflatedBroadcastChannelNotifyStressTest.kt]: Start imports should be used (can be configured once, then do "optimize import" on files that were changed by commit)

File: reactive/kotlinx-coroutines-reactor/test/MonoTest.kt [hunk 3]
```diff
@@ -52,8 +51,8 @@ class MonoTest : TestBase() {
             expectUnreached()
         }, { error ->
             expect(5)
-            assertThat(error, IsInstanceOf(RuntimeException::class.java))
-            assertThat(error.message, IsEqual("OK"))
+            assert(error is RuntimeException)
```
Comment (elizarov) [inline reactive/kotlinx-coroutines-reactor/test/MonoTest.kt]: that should be replaced with `assertTrue(...)`, not `assert(...)`

File: reactive/kotlinx-coroutines-reactor/test/SchedulerTest.kt [hunk 4]
```diff
@@ -24,13 +21,13 @@ class SchedulerTest : TestBase() {
         val mainThread = Thread.currentThread()
         withContext(Schedulers.single().asCoroutineDispatcher()) {
             val t1 = Thread.currentThread()
-            assertThat(t1, IsNot(IsEqual(mainThread)))
+            assert(t1 != mainThread)
```
Comment (elizarov) [inline reactive/kotlinx-coroutines-reactor/test/SchedulerTest.kt]: Should use `assertNotSame(....)` here.

File: kotlinx-coroutines-core/jvm/test/channels/ActorTest.kt [hunk 5]
```diff
@@ -28,14 +27,14 @@ class ActorTest(private val capacity: Int) : TestBase() {
             expect(3)
         }
         actor as Job // type assertion
-        assertThat(actor.isActive, IsEqual(true))
-        assertThat(actor.isCompleted, IsEqual(false))
-        assertThat(actor.isClosedForSend, IsEqual(false))
+        assertEquals(true, actor.isActive)
```
Comment (dkhalanskyjb) [inline kotlinx-coroutines-core/jvm/test/channels/ActorTest.kt]: These are also easy to replace with `assertTrue` and `assertFalse`. Would it be better?

File: reactive/kotlinx-coroutines-reactor/test/FluxTest.kt (hunk 1 above)
Comment (dkhalanskyjb) [reply reactive/kotlinx-coroutines-reactor/test/FluxTest.kt]: Do you mean that `assertEquals` should be taken from `kotlin.test` or that `Assert` should be part of the import at the top of the file?

File: kotlinx-coroutines-core/jvm/test/channels/ConflatedBroadcastChannelNotifyStressTest.kt (hunk 2 above)
Comment (dkhalanskyjb) [reply kotlinx-coroutines-core/jvm/test/channels/ConflatedBroadcastChannelNotifyStressTest.kt]: I've never heard of "start imports". Google suggests that it meant "starred imports". Did it guess right?

**Comment by dkhalanskyjb** [comment]
I think I fixed the problems. I also automatically replaced `assertEquals(null, X)` with `assertNull(X)` and changed uses of `org.junit.Assert` to `kotlin.test`. I have doubts about `ConvertTest` though.

File: kotlinx-coroutines-core/jvm/test/channels/ConflatedBroadcastChannelNotifyStressTest.kt (hunk 2 above)
Comment (elizarov) [reply kotlinx-coroutines-core/jvm/test/channels/ConflatedBroadcastChannelNotifyStressTest.kt]: Yes. That was type. I meant "star imports" as in `import kotlin.test.*`

**Pull request merged**